dkpro-core-io-brat-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/brat/BratWriter.java

```diff
@@ -306,36 +336,41 @@ private void writeAnnotations(JCas aJCas)
 
         // Go through all the annotations but only handle the ones that have no references to
         // other annotations.
+        for (String currentType: spanTypes.keySet()){
+             for (FeatureStructure fs : selectAll(aJCas)) {
+                String typeName=fs.getType().getName();
+                String superType = fs.getCAS().getTypeSystem().getParent(fs.getType()).getName();
+                // Skip excluded types
+                if (excludeTypes.contains(typeName) || excludeTypes.contains(superType) ) {
+                    getLogger().debug("Excluding [" + fs.getType().getName() + "]");
+                    continue;
+                }
+                if (currentType.equalsIgnoreCase(typeName) || currentType.equalsIgnoreCase(superType)) {
+                    writeTextAnnotation(doc, (AnnotationFS) fs);
+                }          
+        	}
+        }
+        // second loop, only for relationships once spans have been set. 
         for (FeatureStructure fs : selectAll(aJCas)) {
             // Skip document annotation
             if (fs == aJCas.getDocumentAnnotationFs()) {
                 continue;
             }
-            
+            String typeName=fs.getType().getName();
+            String superType = fs.getCAS().getTypeSystem().getParent(fs.getType()).getName();
             // Skip excluded types
-            if (excludeTypes.contains(fs.getType().getName())) {
-                getLogger().debug("Excluding [" + fs.getType().getName() + "]");
+            if (excludeTypes.contains(typeName) || excludeTypes.contains(superType) ) {
+                // it would be logged too much times
+            	// getLogger().debug("Excluding [" + fs.getType().getName() + "]");
                 continue;
             }
             
-            if (spanTypes.contains(fs.getType().getName())) {
-                writeTextAnnotation(doc, (AnnotationFS) fs);
+            if (spanTypes.containsKey(typeName) || spanTypes.containsKey(superType)) {
+               // writeTextAnnotation(doc, (AnnotationFS) fs);
             }
-            else if (parsedRelationTypes.containsKey(fs.getType().getName())) {
+            else if (parsedRelationTypes.containsKey(typeName)|| parsedRelationTypes.containsKey(superType)) {
                 relationFS.add(fs);
             }
-            else if (hasNonPrimitiveFeatures(fs) && (fs instanceof AnnotationFS)) {
```

Comment:
@joancf why did you entirely drop the support for annotations with multiple non-primitive features?